### PR TITLE
add sensu version facter (continued)

### PR DESCRIPTION
**Pull request by treydock**
This is a continuation of #818

### CODE REVIEW

File: spec/unit/facter/sensu_version_spec.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  context "on not windows" do
+    before {
+      Facter.clear
+      Facter.fact(:kernel).stubs(:value).returns('NotWindows')
+    }
+    describe 'returns sensu version when present' do
+        it do
+          File.stubs(:exists?).with('/opt/sensu/embedded/bin/sensu-client').returns(true)
+          Facter::Util::Resolution.expects(:exec).with('/opt/sensu/embedded/bin/sensu-client --version 2>&1').returns('0.23.3')
+          expect(Facter.value(:sensu_version)).to eql('0.23.3')
```
Comment (ghoneycutt) [inline spec/unit/facter/sensu_version_spec.rb]: seems these tests are failing

Comment (ghoneycutt) [reply spec/unit/facter/sensu_version_spec.rb]: You can use this as a reference for stubbing facts

https://github.com/ghoneycutt/puppet-module-ssh/blob/master/spec/unit/facter/ssh_spec.rb

**Comment by treydock** [comment]
This is issue with mocha.  Explanation of why be good to use rspec mocks instead of mocha.

https://github.com/puppetlabs/puppetlabs_spec_helper/commit/695876f114a284ad6457e200ecd651824c3b37c7

Change in mocha 1.5.0 that breaks tests:

https://github.com/freerange/mocha/pull/327

Example of rspec mock:

https://github.com/treydock/puppet-osg/blob/master/spec/unit/facter/osg_version_spec.rb

**Comment by treydock** [comment]
@ghoneycutt Fixed tests by using rspec mocks.

**Comment by ghoneycutt** [comment]
Looks good! Could you please squash the commits and keep @bovy89 as a co-author.

https://help.github.com/articles/creating-a-commit-with-multiple-authors/

**Comment by treydock** [comment]
@ghoneycutt Squashed and co-author change made.

**Comment by ghoneycutt** [comment]
Replaces #818

**Comment by ghoneycutt** [comment]
Released in v2.52.0

**Pull request merged**